src/js/components/Posts.js

```diff
@@ -1,103 +1,95 @@
+  constructor(props) {
+    super(props)
 
-        this.notifyError = (which) => {
-            console.warn('Error found in Post section "' + which +'"')
-            this.setState({['error_' + which]: true})
-        }
+    this.state = {
+      posts: [],
+      title: null,
+      moreUrl: null,
+      moreText: null,
     }
+  }
 
+  componentDidMount() {
+    let category = ReactDOM.findDOMNode(this).parentNode.dataset.category
+    this.setState({
+      category,
+      title: ReactDOM.findDOMNode(this).parentNode.dataset.title,
+      moreText: ReactDOM.findDOMNode(this).parentNode.dataset.moreText,
+    })
 
-    render() {
-        if (this.state['error_' + TECH] && this.state['error_' + NON_TECH]) {
-            console.warn('BlogPost section hided')
-            return null
-        }
-
-        return (
-            <aside className="fullWidth" id="ourPosts">
-                <div className="mainContainer">
-                    <header>
-                        <h2>Sharing our journey and work with the world</h2>
-                    </header>
-                    <div className="columns">
-                        <section>
-                            <h3 className="title">Our journey</h3>
-                            <div className="postColumn" id="non-tech-posts">
-                                <Posts kind={NON_TECH} errorHandler={this.notifyError} />
-                            </div>
-                        </section>
-                        <section>
-                            <h3 className="title">Technical posts</h3>
-                            <div className="postColumn" id="tech-posts">
-                                <Posts kind={TECH} errorHandler={this.notifyError} />
-                            </div>
-                        </section>
-                    </div>
-                </div>
-            </aside>
-        )
-    }
-}
-
-class Posts extends Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            state: states.LOADING,
-            posts: []
-        }
-    }
+    loadPosts(category)
+      .then(posts => this.setState({ posts }))
+  }
 
-    componentWillMount() {
-        const loader = this.props.kind == TECH ? loadTechPosts : loadNonTechPosts
-        loader()
-            .then(posts => this.setState({ state: states.LOADED, posts }))
-            .catch(err => {
-                console.error(err)
-                this.setState({ state: states.ERROR })
-                this.props.errorHandler(this.props.kind)
-            })
+  render() {
+    const {posts, category, title, moreText} = this.state
+    if (posts.length == 0) {
+      return <div></div>
     }
 
-    render() {
-        const { state, posts } = this.state
-        if (state === states.LOADING) {
-            return <Loading />
-        } else if (state === states.ERROR) {
-            return <LoadingError />
-        }
+    document.dispatchEvent(events.ACTIVE_EVENT);
```

Comment:
what do you think about using a promise instead of throwing custom events?

```js
let resolveBlogPostsActive, rejectBlogPostsActive
export blogPostsActive = new Promise((resolve, reject) => {
  resolveBlogPostsActive = resolve
  rejectBlogPostsActive = reject
})

// and then
resolveBlogPostsActive()
// or
rejectBlogPostsActive()

// and consume it like
import { blogPostsActive } from 'foo'

blogPostsActive.then(() => {
  // is ok
}).catch(err => {
  // is not ok
})
```